middleware/replacer.go

```diff
@@ -40,6 +41,7 @@ func NewReplacer(r *http.Request, rr *responseRecorder, emptyValue string) Repla
 			"{host}":     r.Host,
 			"{path}":     r.URL.Path,
 			"{query}":    r.URL.RawQuery,
+			"{frag}":     r.URL.Fragment,
```

Comment:
I wouldn't worry about this backward compatibility at this point. Let's just have one way to do it - {fragment}